api/node_pools.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package api
+
+const (
+	// NodePoolAll is the node pool that always includes all nodes.
+	NodePoolAll = "all"
+
+	// NodePoolDefault is the default node pool.
+	NodePoolDefault = "default"
+)
+
+// NodePools is used to access node pools endpoints.
+type NodePools struct {
+	client *Client
+}
+
+func (c *Client) NodePools() *NodePools {
+	return &NodePools{client: c}
+}
+
+func (n *NodePools) List(q *QueryOptions) ([]*NodePool, *QueryMeta, error) {
+	var resp []*NodePool
+	qm, err := n.client.query("/v1/node/pools", &resp, q)
+	if err != nil {
+		return nil, nil, err
+	}
+	return resp, qm, nil
+}
+
+func (n *NodePools) PrefixList(prefix string, q *QueryOptions) ([]*NodePool, *QueryMeta, error) {
+	if q == nil {
+		q = &QueryOptions{}
+	}
+	q.Prefix = prefix
+	return n.List(q)
+}
+
+func (n *NodePools) Info(name string, q *QueryOptions) (*NodePool, *QueryMeta, error) {
+	var resp NodePool
+	qm, err := n.client.query("/v1/node/pool/"+name, &resp, q)
```

Comment:
Oh I see. Most of the methods in `api` I had seen have not validation, but some do, which is nice.

Reading the docs, `url.PathEscape` already handles slashes, albeit by replacing them with their percent representation rather than removing, but I think that's OK in this case, so I didn't use `cleanPathString`.